packages/medusa/src/services/order-edit.ts

```diff
@@ -315,6 +322,130 @@ export default class OrderEditService extends TransactionBaseService {
     })
   }
 
+  /**
+   * Create or update order edit item change line item and apply the quatity
+   * - If the item change already exists then update the quantity of the line item as well as the line adjustments and tax lines
+   * - If the item change does not exist then create the item change and cloned the original line item with the quantity as well as the line adjustments and tax lines
+   * @param orderEditId
+   * @param originalLineItemId
+   * @param data
+   */
+  async updateLineItem(
+    orderEditId: string,
+    originalLineItemId: string,
+    data: { quantity: number }
+  ): Promise<void> {
+    return await this.atomicPhase_(async (manager) => {
+      const orderEdit = await this.retrieve(orderEditId, {
+        select: [
+          "id",
+          "items",
+          "order_id",
+          "created_at",
+          "requested_at",
+          "confirmed_at",
+          "declined_at",
+          "canceled_at",
+        ],
+        relations: ["items"],
```

Comment:
suggestion: instead of getting all the items couldn't we just fetch the item to change directly from the `lineItemService` and then check `item.order_edit_id === orderEditId`?